python/cudf/cudf/core/indexed_frame.py

```diff
@@ -526,3 +532,208 @@ def _n_largest_or_smallest(self, largest, n, columns, keep):
             return self.take(indices, keep_index=True)
         else:
             raise ValueError('keep must be either "first", "last"')
+
+    def _align_to_index(
+        self: T,
+        index: ColumnLike,
+        how: str = "outer",
+        sort: bool = True,
+        allow_non_unique: bool = False,
+    ) -> T:
+        index = cudf.core.index.as_index(index)
+
+        if self.index.equals(index):
+            return self
+        if not allow_non_unique:
+            if len(self) != len(self.index.unique()) or len(index) != len(
+                index.unique()
+            ):
+                raise ValueError("Cannot align indices with non-unique values")
+
+        lhs = cudf.DataFrame._from_data(self._data, index=self.index)
+        rhs = cudf.DataFrame(index=index)
+
+        tmp_col_id = str(uuid4())
```

Comment:
A comment on why this temp col is necessary (for sorting) would be helpful. A more meaningful name e.g. sorted_index_col might also be helpful.